File: ios/App/App/PushPermissionPlugin.m
#import <Foundation/Foundation.h>
#import <Capacitor/Capacitor.h>

CAP_PLUGIN(PushPermissionPlugin, "PushPermission",
    CAP_PLUGIN_METHOD(requestPermission, CAPPluginReturnPromise);
    CAP_PLUGIN_METHOD(checkPermission, CAPPluginReturnPromise);
)

File: ios/App/App/PushPermissionPlugin.swift
import Foundation
import Capacitor
import OneSignalFramework

@objc(PushPermissionPlugin)
public class PushPermissionPlugin: CAPPlugin {
    
    @objc func requestPermission(_ call: CAPPluginCall) {
        DispatchQueue.main.async {
            OneSignal.Notifications.requestPermission({ accepted in
                NSLog("OneSignal permission accepted: \(accepted)")
                call.resolve([
                    "accepted": accepted
                ])
            }, fallbackToSettings: true)
        }
    }
    
    @objc func checkPermission(_ call: CAPPluginCall) {
        DispatchQueue.main.async {
            let permission = OneSignal.Notifications.permission
            call.resolve([
                "permission": permission
            ])
        }
    }
}